Sources/VLCActionSheet.swift

```diff
@@ -0,0 +1,422 @@
+            collectionViewHeightConstraint.constant = CGFloat(dataSource.numberOfRows()) * cellHeight
+            maxCollectionViewHeightConstraint.constant = presentingViewController.view.frame.size.height / 2
+            collectionView.setNeedsLayout()
+            collectionView.layoutIfNeeded()
+        }
+    }
+
+    @available(iOS 11.0, *)
+    override func viewSafeAreaInsetsDidChange() {
+        bottomBackgroundViewHeightConstraint.constant = view.safeAreaInsets.bottom
+    }
+
+    // MARK: UIViewController
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.addSubview(backgroundView)
+        view.addSubview(mainStackView)
+
+        // wip: Currently adapting to new UI
+        //       option to hide cancelbutton
+        mainStackView.addArrangedSubview(headerView)
+        mainStackView.addArrangedSubview(collectionView)
+//        mainStackView.addArrangedSubview(cancelButton)
+        mainStackView.addArrangedSubview(bottomBackgroundView)
+
+        backgroundView.frame = UIScreen.main.bounds
+
+        setupMainStackViewConstraints()
+        setupCollectionViewConstraints()
+//        setupCancelButtonConstraints()
+        setuplHeaderViewConstraints()
+        setupBottomBackgroundView()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        mainStackView.isHidden = true
+        updateViewConstraints()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        // This is to avoid a horrible visual glitch!
+        mainStackView.isHidden = false
+
+        UIView.transition(with: backgroundView, duration: 0.2, options: .transitionCrossDissolve, animations: { [weak self] in
+            self?.backgroundView.isHidden = false
+            }, completion: nil)
+
+        let realMainStackView = mainStackView.frame
+
+        mainStackView.frame.origin.y += mainStackView.frame.origin.y
+
+        UIView.animate(withDuration: 0.6, delay: 0, usingSpringWithDamping: 0.8, initialSpringVelocity: 0, options: .curveEaseOut, animations: {
+            [mainStackView] in
+            mainStackView.frame = realMainStackView
+            }, completion: nil)
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        coordinator.animate(alongsideTransition: { [weak self] _ in
+            self?.maxCollectionViewHeightConstraint.constant = size.height / 2
+            self?.collectionView.layoutIfNeeded()
+        })
+    }
+
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+        collectionView.collectionViewLayout.invalidateLayout()
+    }
+
+    @objc func addAction(closure action: @escaping (_ item: Any) -> Void) {
+        self.action = action
```

Comment:
it sets the action and doesn’t „add“ the action.

Reply:
Indeed, at first I wanted it to be an array of actions :D